go.mod

```diff
@@ -12,13 +12,13 @@ require (
 	github.com/hashicorp/terraform-plugin-log v0.9.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.33.0
 	github.com/mitchellh/mapstructure v1.5.0
-	github.com/polytomic/polytomic-go v0.2.1-0.20240416175622-09865140ff34
+	github.com/polytomic/polytomic-go v1.0.0
```

Comment:
I don't think it's safe to do this update to `v1.0.0`, `v0.2.1-0.20240416175622-09865140ff34` is branched since we need to manually add `tfsdk` tags to the structs (https://github.com/polytomic/polytomic-go/blob/main/types.go)